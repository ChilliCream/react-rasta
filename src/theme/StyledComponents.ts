import {ComponentClass, StatelessComponent} from "react";
import * as styledComponents from "styled-components";
import Theme from "./Theme";

const {
  default: styled,
  css,
  createGlobalStyle,
  keyframes,
  ThemeProvider,
} = styledComponents as styledComponents.ThemedStyledComponentsModule<Theme>;

export default styled;
export {css, createGlobalStyle, keyframes, ThemeProvider};
