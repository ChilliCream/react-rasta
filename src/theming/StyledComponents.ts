// These imports are needed for declaration files
import * as _React from "react";
import * as styledComponents from "styled-components";
// -------------------------------------------------------------------
import { ThemedStyledComponentsModule } from "styled-components";
import Theme from "./Theme";

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  withTheme,
  ThemeProvider
} = styledComponents as ThemedStyledComponentsModule<Theme>;

export default styled;
export { css, injectGlobal, keyframes, withTheme, ThemeProvider };
