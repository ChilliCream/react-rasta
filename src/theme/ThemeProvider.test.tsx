import {mount} from "enzyme";
import "jest-styled-components";
import React from "react";
import {Column, Container, Row} from "..";
import {ThemeProvider} from "./StyledComponents";

describe("<ThemeProvider />", () => {
  it("should match the snapshot (gutterWidth set to 100)", () => {
    // act
    const result = mount(
      <ThemeProvider theme={{gutterWidth: 100}}>
        <Container>
          <Row>
            <Column>Test</Column>
          </Row>
        </Container>
      </ThemeProvider>,
    );

    // assert
    expect(result).toMatchSnapshot();
  });
});
