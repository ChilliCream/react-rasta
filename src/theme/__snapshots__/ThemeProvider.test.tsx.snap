// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<ThemeProvider /> should match the snapshot (gutterWidth set to 100) 1`
] = `
.c3 {
  position: relative;
  width: 100%;
  min-height: 1px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -50px;
  margin-left: -50px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 > .c2 {
  padding-right: 50px;
  padding-left: 50px;
}

.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  padding-right: 50px;
  padding-left: 50px;
}

@media (min-width:576px) {
  .c0 {
    max-width: 540px;
  }
}

@media (min-width:768px) {
  .c0 {
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c0 {
    max-width: 960px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1140px;
  }
}

<ThemeProvider
  theme={
    Object {
      "gutterWidth": 100,
    }
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <styled.div>
            <div
              className="c2 c3"
            >
              Test
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</ThemeProvider>
`;
